From 5dd9cee8d4bca16e71bddfcac833ae93fb653cd6 Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <xiaowen.liu@nxp.com>
Date: Mon, 22 Feb 2016 16:44:10 -0500
Subject: [PATCH 631/635] MA-7610 Coverity 26614: RESOURCE_LEAK in Component
 Other.

The file handle is leaked when run into exception code.
Close file handle before return in exception.

Signed-off-by: Xiaowen Liu <xiaowen.liu@nxp.com>
---
 mx6/libcamera3/Ov5640Csi.cpp  | 3 ++-
 mx6/libcamera3/Ov5640Mipi.cpp | 3 ++-
 mx6/libcamera3/Ov5642Csi.cpp  | 3 ++-
 mx6/libcamera3/TVINDevice.cpp | 5 ++++-
 mx6/libcamera3/UvcDevice.cpp  | 1 +
 5 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/mx6/libcamera3/Ov5640Csi.cpp b/mx6/libcamera3/Ov5640Csi.cpp
index b8a1c77..8cff148 100644
--- a/mx6/libcamera3/Ov5640Csi.cpp
+++ b/mx6/libcamera3/Ov5640Csi.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2016 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -67,6 +67,7 @@ status_t Ov5640Csi::initSensorStaticData()
     mSensorFormatCount = changeSensorFormats(sensorFormats, mSensorFormats, index);
     if (mSensorFormatCount == 0) {
         ALOGE("%s no sensor format enum", __func__);
+        close(fd);
         return BAD_VALUE;
     }
 
diff --git a/mx6/libcamera3/Ov5640Mipi.cpp b/mx6/libcamera3/Ov5640Mipi.cpp
index 5915642..4689317 100644
--- a/mx6/libcamera3/Ov5640Mipi.cpp
+++ b/mx6/libcamera3/Ov5640Mipi.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2016 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -68,6 +68,7 @@ status_t Ov5640Mipi::initSensorStaticData()
     mSensorFormatCount = changeSensorFormats(sensorFormats, mSensorFormats, index);
     if (mSensorFormatCount == 0) {
         ALOGE("%s no sensor format enum", __func__);
+        close(fd);
         return BAD_VALUE;
     }
 
diff --git a/mx6/libcamera3/Ov5642Csi.cpp b/mx6/libcamera3/Ov5642Csi.cpp
index fbb75b2..9cd2580 100644
--- a/mx6/libcamera3/Ov5642Csi.cpp
+++ b/mx6/libcamera3/Ov5642Csi.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2016 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -67,6 +67,7 @@ status_t Ov5642Csi::initSensorStaticData()
     mSensorFormatCount = changeSensorFormats(sensorFormats, mSensorFormats, index);
     if (mSensorFormatCount == 0) {
         ALOGE("%s no sensor format enum", __func__);
+        close(fd);
         return BAD_VALUE;
     }
 
diff --git a/mx6/libcamera3/TVINDevice.cpp b/mx6/libcamera3/TVINDevice.cpp
index f7df8e0..09c6652 100644
--- a/mx6/libcamera3/TVINDevice.cpp
+++ b/mx6/libcamera3/TVINDevice.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2016 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -53,11 +53,13 @@ status_t TVINDevice::initSensorStaticData()
         ALOGI("%s Get current mode: NTSC", __func__);
     else {
         ALOGE("%s Error!Get invalid mode: %llu", __func__, mSTD);
+        close(fd);
         return BAD_VALUE;
     }
 
     if (ioctl(fd, VIDIOC_S_STD, &mSTD) < 0) {
         ALOGE("%s VIDIOC_S_STD failed", __func__);
+        close(fd);
         return BAD_VALUE;
     }
 
@@ -75,6 +77,7 @@ status_t TVINDevice::initSensorStaticData()
     mSensorFormatCount = changeSensorFormats(sensorFormats, mSensorFormats, index);
     if (mSensorFormatCount == 0) {
         ALOGE("%s no sensor format enum", __func__);
+        close(fd);
         return BAD_VALUE;
     }
 
diff --git a/mx6/libcamera3/UvcDevice.cpp b/mx6/libcamera3/UvcDevice.cpp
index 807c317..7ee47e2 100644
--- a/mx6/libcamera3/UvcDevice.cpp
+++ b/mx6/libcamera3/UvcDevice.cpp
@@ -84,6 +84,7 @@ status_t UvcDevice::initSensorStaticData()
     mSensorFormatCount = changeSensorFormats(sensorFormats, mSensorFormats, index);
     if (mSensorFormatCount == 0) {
         ALOGE("%s no sensor format enum", __func__);
+        close(fd);
         return BAD_VALUE;
     }
 
-- 
1.8.0

